receiver/snmpreceiver/scraper.go

```diff
@@ -39,13 +66,339 @@ func newScraper(logger *zap.Logger, cfg *Config, settings component.ReceiverCrea
+// using the current index and attribute config to access the correct value
+func getIndexedDataPointAttributes(
+	configHelper *configHelper,
+	columnOID string,
+	indexString string,
+	columnOIDIndexedAttributeValues map[string]indexedAttributeValues,
+) (map[string]string, error) {
+	datapointAttributes := map[string]string{}
+
+	for _, attribute := range configHelper.getMetricConfigAttributes(columnOID) {
+		attributeName := attribute.Name
+
+		attributeKey := attributeName
+		// Use alternate attribute key if available
+		if value := configHelper.getAttributeConfigValue(attributeKey); value != "" {
+			attributeKey = value
+		}
+
+		var attributeValue string
+		prefix := configHelper.getAttributeConfigIndexedValuePrefix(attributeName)
+		oid := configHelper.getAttributeConfigOID(attributeName)
+		switch {
+		case prefix != "":
+			attributeValue = prefix + indexString
+		case oid != "":
+			attributeValue = columnOIDIndexedAttributeValues[oid][indexString]
+		default:
+			attributeValue = attribute.Value
+		}
+
+		// If no good attribute value could be found
+		if attributeValue == "" {
+			return nil, errors.New(errMsgAttributeEmptyValue)
+		}
+		datapointAttributes[attributeKey] = attributeValue
+	}
+
+	return datapointAttributes, nil
+}
+
+// getResourceAttributes creates a map of key/values for all related resource attributes. Keys
+// will come directly from the metric config's resource attribute values. Values will come
+// from the related attribute config's prefix value plus the index OR the previously collected
+// resource attribute indexed data.
+func getResourceAttributes(
+	configHelper *configHelper,
+	columnOID string,
+	indexString string,
+	columnOIDIndexedResourceAttributeValues map[string]indexedAttributeValues,
+) (map[string]string, error) {
+	resourceAttributes := map[string]string{}
+
+	for _, attributeName := range configHelper.getResourceAttributeNames(columnOID) {
+		prefix := configHelper.getResourceAttributeConfigIndexedValuePrefix(attributeName)
+		oid := configHelper.getResourceAttributeConfigOID(attributeName)
+		switch {
+		case prefix != "":
+			resourceAttributes[attributeName] = prefix + indexString
+		case oid != "":
+			attributeValue := columnOIDIndexedResourceAttributeValues[oid][indexString]
+
+			if attributeValue == "" {
+				return nil, errors.New(errMsgResourceAttributeEmptyValue)
+			}
+
+			resourceAttributes[attributeName] = attributeValue
+		default:
+			return nil, errors.New(errMsgResourceAttributeEmptyValue)
+		}
+	}
+
+	return resourceAttributes, nil
+}
+
+// scrapeIndexedAttributes retrieves all SNMP data from attribute (or resource attribute)
+// config column OIDs and stores the returned indexed data for later use by metrics
+func (s *snmpScraper) scrapeIndexedAttributes(
+	columnOIDs []string,
+	scraperErrors *scrapererror.ScrapeErrors,
+) map[string]indexedAttributeValues {
+	indexedAttributeValues := map[string]indexedAttributeValues{}
```

Comment:
Let's disambiguate the variable name from the struct type. There are a number of places throughout the file where this is needed.